src/app/core/layout/header/header.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterLink } from '@angular/router';
+import { AuthenticationService } from '../../services/authentication.service';
+import { MmsaiAccountButtonComponent } from '../../components/mmsai-account-button/mmsai-account-button.component';
+import { MenuItem } from 'primeng/api';
+import { Store } from '@ngrx/store';
+import { signOut } from '../../states/auth/auth.actions';
+
+@Component({
+  selector: 'app-header',
+  standalone: true,
+  imports: [CommonModule, RouterLink, MmsaiAccountButtonComponent],
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  isLoggedIn!: boolean;
+  displayName?: string | null;
+  menuItems!: MenuItem[];
+
+  constructor(
+    private authenticationService: AuthenticationService,
+    private store: Store,
+  ) {}
+
+  ngOnInit() {
+    this.authenticationService.getUser().subscribe((user) => {
+      if (user) {
+        this.isLoggedIn = true;
+        this.displayName = user.displayName ? user.displayName : user.email;
```

Comment:
ternary operator is very hard to read. You should drop using it. Your code gets more readable. In this case you want to assign a default value. So you can use the [Nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) operator. It's made for that job 🤩

```suggestion
        this.displayName = user.displayName ?? user.email;
```